File: targets/python-flask/routes/files.py
"""File download routes."""
from flask import Blueprint, request, jsonify

from services import file_service

files_bp = Blueprint("files", __name__, url_prefix="/files")


@files_bp.route("/download")
def download():
    name = request.args.get("name", "")
    content = file_service.read_user_file(name)
    return jsonify({"content": content})


@files_bp.route("/download_safe")
def download_safe():
    name = request.args.get("name", "")
    try:
        content = file_service.read_whitelisted(name)
    except ValueError:
        return jsonify({"error": "file not allowed"}), 400
    return jsonify({"content": content})

File: targets/js-ts-express/routes/files.ts
// File download routes.
import { Router, Request, Response } from 'express';
import { readUserFile, readWhitelisted } from '../services/fileService';

const router = Router();

router.get('/download', (req: Request, res: Response) => {
  const name = String(req.query.name ?? '');
  res.json({ content: readUserFile(name) });
});

router.get('/download_safe', (req: Request, res: Response) => {
  const name = String(req.query.name ?? '');
  try {
    res.json({ content: readWhitelisted(name) });
  } catch {
    res.status(400).json({ error: 'file not allowed' });
  }
});

export default router;
